docs/concepts/transactions.md

```diff
@@ -0,0 +1,63 @@
+
+* **Route** returns a string that specifies which module this message is a part of and, thus, which module contains the handler used to implement the actions of this message. Applications that inherit from [baseapp](./baseapp.md) use this function to help deliver transactions by invoking the correct handler(s). For example, `MsgSend` is defined in and handled by the `bank` module; its `Route` function returns the name of the `bank` module.
+* **Type** returns a short, human-readable string that describes what the message does. For example, `MsgSend`'s type is `"send"`.
+* **ValidateBasic** implements a set of stateless sanity checks for the message and returns an `Error`. For example, the `validateBasic` method for `MsgSend`, the message which sends coins from one account to another, checks that both sender and recipient addresses are valid and the amount is nonnegative.
+* **GetSignBytes** returns a `[]byte` representation of the message which is used by the signer to sign it.
+* **GetSigners** returns a list of addresses whose corresponding signatures are required for the message to be valid. For example, `MsgSend` requires a signature from the sender of the coins.
+
+While messages contain the information for state transition logic, a transaction's other metadata and relevant information are stored in the `TxBuilder` and `CLIContext`. 
+
+## Transaction Generation
+
+[`Contexts`](https://godoc.org/context) are immutable data structures that can be cloned and updated as it is passed between processes and request handlers. In the process of creating a transaction, two contexts are created: the `CLIContext` and `TxBuilder`. Both are automatically generated and do not need to be defined by the developer, but do require input from the transaction creator (e.g. using flags through the CLI). The `TxBuilder` contains data such as the `TxEncoder` defined by the developer used to encode messages, the [`KeyBase`](https://github.com/cosmos/cosmos-sdk/blob/85ebf5f72ea21f926b9f371b5dae6af65292c02d/crypto/keys/types.go#L14-L60) used to manage keys, as well as `fees`, `gasPrices`, and other user-provided values. The `CLIContext` is initialized using the application's `codec` and data more closely related to the user interaction, such as which account the transaction is originating from. 
+
+Every message in a transaction must be signed by the addresses specified by `GetSigners`. The signing process is handled by the [`auth`](https://github.com/cosmos/cosmos-sdk/tree/67f6b021180c7ef0bcf25b6597a629aca27766b8/docs/spec/auth) module and automatically performed when the transaction is created, unless the user choses to generate and sign separately. The `TxBuilder` contains the `KeyBase` used to manage keys, and thus used to perform the signing operations. The `CLIContext` is used to broadcast a transaction.
```

Comment:
The signing process must be handled by a module. The most widely used module for that purpose is the `auth` module